### DISCUSSION TITLE
Is it possible to generate schema from openapi json file

### DISCUSSION THREAD

**Discussion by hari-nair-on** [loopbackio/loopback-next#8182] [Q&A]
Is it possible to generate a schema from a openapi swagger json file.

**Comment by achrinza** [comment]
Hi @hari-nair-on, could you clarify what kind of schema you'd like to generate?

The LoopBack 4 CLI can generate an OpenAPI client in an existing LoopBack 4 project using `lb4 openapi --client`. This is useful for linking a LoopBack 4 project to existing OpenAPI 2/3.0-documented APIs.

For generating an OpenAPI 3.0 spec of the current LoopBack 4 project: By default, the `/openapi.json` is exposed when the server is running, and `npm run openapi-spec` can be used to directly export it to a file.

**Reply by nairgh** [reply]
Thanks for the reply. Really appreciated.  I am using loopback 4. Created a datasource [lb4 datasource] and  poiinted towards a json file which is created by openapi spec. I  am trying to create a scehema  using that json file either wholly or partially.  What is the best way to do this.  To show a controller for crud we need a model right. How can i create a model out of this. The json file already have valid fields to create a model. It is possibly a tedious task if the json file is large enough. Any practical idea how can i achieve this.
